### ISSUE TITLE
How can one add weights for different tags?

### ISSUE THREAD

**Issue by liebkne** [kmkurn/pytorch-crf#47]

**Comment by kmkurn** [comment]
Hi, could you please explain a bit more clearly about this?

**Comment by isolet** [comment]
The document of `nn.CrossEntropyLoss` says 

> It is useful when training a classification problem with C classes. If provided, the optional argument weight should be a 1D Tensor assigning weight to each of the classes. This is particularly useful when you have an unbalanced training set.

I wonder if this is possible in the CRF loss layer. Thanks.

**Comment by kmkurn** [comment]
I see. The CRF does not offer this, but there might be tricks to do this with the current implementation, depending on how you want to weight the tags. For instance, you may incorporate the weights into the emission scores before feeding them into the CRF layer. You need to work out the maths and check if this can achieve the weighting you want.

**Comment by allenyummy** [comment]
Hey, 
Did you find out any solution to add weights for tags ?

Thanks.

**Comment by kmkurn** [comment]
Hi @allenyummy. It's been a while since my last post. What you wrote seems to be what I had in mind, but you'll need to verify yourself if it achieves what you want.
